### ISSUE TITLE
Exception when passing legend_out=False to relplot in seaborn 0.11.0

### ISSUE THREAD

**Issue by haydenflinner** [mwaskom/seaborn#2362]
```
Traceback (most recent call last):
  File "my.py", line 156, in <module>
    g = sns.relplot(data=thishost,
  File "/spare/local/hflinner/.conda/envs/plotting/lib/python3.8/site-packages/seaborn/_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "/spare/local/hflinner/.conda/envs/plotting/lib/python3.8/site-packages/seaborn/relational.py", line 1030, in relplot
    g.add_legend(legend_data=p.legend_data,
  File "/spare/local/hflinner/.conda/envs/plotting/lib/python3.8/site-packages/seaborn/axisgrid.py", line 170, in add_legend
    adjust_legend_subtitles(figlegend)
UnboundLocalError: local variable 'figlegend' referenced before assignment
```

call:
```
    g = sns.relplot(data=thishost,
                   ...
                    facet_kws={'legend_out': False})
```

My hack fix is an axisgrid.py edit of the below. I don't know if this is what the code is intended to do, but it at least runs and my legend is now within the graphic:
```
 164         else:
 165             # Draw a legend in the first axis
 166             ax = self.axes.flat[0]
 167             kwargs.setdefault("loc", "best")
 168 
 169             leg = ax.legend(handles, labels, **kwargs)
 170             leg.set_title(title, prop={"size": title_size})
 171             if adjust_subtitles:
 172                 adjust_legend_subtitles(leg)
 173 
 174             self._legend = leg
```

Additionally, this had the unexpected but pleasant surprise fix of allowing my custom Major/Minor tick locators to take effect. I had tried every incantation under the sun, from changing matplotlib rcparams to make sure ticks were visible, to overriding the seaborns style, nothing would let me show minor ticks or customize the major. I did manage to get different major ticks, i.e. I saw the grid within the graphic change, but the labels were still only one the ones seaborn wanted to label, not all the major ticks. With legend_out=False and the above minor fix, suddenly all of my tick overrides work. Pasting them below just in case there's a known issue:

```
    # Seaborns seems to somehow be breaking the LogLocator and FixedLocator from doing what they're supposed to do
    from matplotlib.ticker import LogLocator, NullFormatter, FixedLocator
    # locmaj = LogLocator(base=10,numticks=15) 
    locmaj = FixedLocator(yticks)
    ax.yaxis.set_major_locator(locmaj)
    # locmin = LogLocator(base=10,numticks=15) 
    locmin = LogLocator(base=10,subs=(0.2,0.4,0.6,0.8),numticks=15)
    ax.yaxis.set_minor_locator(locmin)
```

**Comment by mwaskom** [comment]
I think this is fixed on master by #2304

**Comment by mwaskom** [comment]
Not sure what you're getting at with the locator/ticks but if you have a reproducible example demonstrating a bug, please open a separate issue.
